components/net/fetch/methods.rs

```diff
@@ -498,3 +530,94 @@ fn is_null_body_status(status: &Option<StatusCode>) -> bool {
         _ => false
     }
 }
+
+/// https://fetch.spec.whatwg.org/#should-response-to-request-be-blocked-due-to-nosniff?
+#[inline]
+fn should_block_nosniff(request: &Request, response: &Response) -> bool {
+    header! { (XContentTypeOptions, "X-Content-Type-Options") => [String] };
+
+    // Step 2
+    if let Some(header) = response.headers.get::<XContentTypeOptions>() {
```

Comment:
Let's do:
```rust
let header = match response.headers.get::<XContentTypeOptions>() {
    Some(h) => h,
    None => return false,
};
```
to avoid the extra level of indentation.